### DISCUSSION TITLE
Auth Policy Blocking Inserts

### DISCUSSION THREAD

**Discussion by jaredcosulich** [supabase/supabase#6069] [Questions]
I have an auth policy for insert on projects that is just set to true but I am still being stopped from creating a project via the supabase client.

Here are the policies:
<img width="1093" alt="Screen Shot 2022-03-25 at 11 13 26 AM" src="https://user-images.githubusercontent.com/191799/160148582-8b0bbd23-ed02-4943-a9c7-23ed0b01fe4f.png">

but when I try to create a project I get the error:

```
new row violates row-level security policy for table \"projects\"
```

Are the other policies for select, update, or delete impacting the insert command?

**Comment by GaryAustin1** [comment]
"By default, every time you run insert(), the client library will make a select to return the full record. This is convenient, but it can also cause problems if your Policies are not configured to allow the select operation. If you are using Row Level Security and you are encountering problems, try setting the returning param to minimal."
https://supabase.com/docs/reference/javascript/insert#notes

**Reply by jaredcosulich** [reply]
Thank you. What does "try setting the returning param to minimal" mean? How / where do I do that exactly?

**Reply by GaryAustin1** [reply]
insert([{ key: 'value' }], { returning: 'minimal' })

**Reply by jaredcosulich** [reply]
Nevermind. I found it in the documentation. It's a second parameter on the insert function:

```
const { body, error } = await supabaseClient
    .from('projects')
    .insert({
      team_id: team.id,
      title: "New Project",
      ...data
    }, {
      returning: 'minimal'
    })
```

**Reply by jaredcosulich** [reply]
Thanks!
